### PR TITLE
EmptyElseBlock

### PR DESCRIPTION
**Pull request by IvenBach**
Addresses issues #2909, #3031, #3037. Includes EmptyElseBlockQuickFix.

### CODE REVIEW

File: Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            _state = state;
+        }
+
+        public void Fix(IInspectionResult result)
+        {
+            IModuleRewriter rewriter = _state.GetRewriter(result.QualifiedSelection.QualifiedName);
+
+            //dynamic used since it's not known at run-time
+            UpdateContext((dynamic)result.Context, rewriter);
```
Comment (retailcoder) [inline Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs]: If it's going to be passed as a `VBAParser.ElseBlockContext`, might as well cast it to `VBAParser.ElseBlockContext` ;-)

File: Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        private void UpdateContext(VBAParser.ElseBlockContext context, IModuleRewriter rewriter)
+        {
+            VBAParser.BlockContext elseBlock = context.block();
+
+            if (elseBlock.ChildCount == 0 )
+            {
+                //string rewrittenBlock = AdjustedBlockText(context.block());
+                //rewriter.InsertBefore(context.start.StartIndex, rewrittenBlock);
```
Comment (retailcoder) [inline Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs]: Please remove commented-out code

File: Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+             * only the else block thats empty. IE: it doesn't affect
+             * the TRUE portion that preceeds it.
+             */
+        }
+
+        //private bool FirstBlockStmntHasLabel(VBAParser.BlockContext block)
+        //    => block.blockStmt()?.FirstOrDefault()?.statementLabelDefinition() != null;
+
+        //private bool BlockHasDeclaration(VBAParser.BlockContext block)
+        //    => block.blockStmt()?.Any() ?? false;
```
Comment (retailcoder) [inline Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs]: dead code, kill :)

File: Rubberduck.Parsing/Inspections/Resources/InspectionsUI.resx [hunk 4]
```diff
@@ -766,4 +766,16 @@ If the parameter can be null, ignore this inspection result; passing a null valu
   <data name="CodeInspectionSeverity_Warning" xml:space="preserve">
     <value>Warning</value>
   </data>
+  <data name="EmptyElseBlockInspectionResultFormat" xml:space="preserve">
+    <value>Else block has no executable statements</value>
```
Comment (retailcoder) [inline Rubberduck.Parsing/Inspections/Resources/InspectionsUI.resx]: "has" is a bit weak.. perhaps "Else block *contains* no executable statements"?

File: Rubberduck.Parsing/Inspections/Resources/InspectionsUI.resx [hunk 5]
```diff
@@ -766,4 +766,16 @@ If the parameter can be null, ignore this inspection result; passing a null valu
...
+  </data>
+  <data name="EmptyElseBlockInspectionMeta" xml:space="preserve">
+    <value>An Else block should have executable statements</value>
```
Comment (retailcoder) [inline Rubberduck.Parsing/Inspections/Resources/InspectionsUI.resx]: The meta string is where we get to explain the rationale behind an inspection. Could be expanded a bit (with punctuation!).

> An empty else block serves no purpose and can be confusing for the maintainer, which is left wondering whether that execution path was intended to do something at one point. Clear code only contains executable execution paths.

Something like that.

File: Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs (hunk 2 above)
Comment (IvenBach) [inline Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs]: Is the use of `ChildCount == 0` the best way of doing this? I wasn't sure what else to use and this seemed appropriate.

File: RetailCoder.VBE/UI/FindSymbol/FindSymbolControl.xaml.cs [hunk 6]
```diff
@@ -37,7 +37,7 @@ private void CommandBinding_OnCanExecute(object sender, CanExecuteRoutedEventArg
 
         private void UIElement_OnPreviewKeyDown(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Enter && ViewModel.CanExecute())
+            if (e.Key == Key.Enter && ViewModel.CanExecute())
```
Comment (Hosch250) [inline RetailCoder.VBE/UI/FindSymbol/FindSymbolControl.xaml.cs]: Undo this.

File: Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs (hunk 2 above)
Comment (Hosch250) [inline Rubberduck.Inspections/QuickFixes/RemoveEmptyElseBlockQuickFix.cs]: Looks OK.

**Comment by Hosch250** [comment]
I think you need to handle single-line else's too. Not sure.

**Comment by retailcoder** [comment]
Please sync & resolve conflicts

**Pull request closed without merging**